### DISCUSSION TITLE
Classic mirror queue does not deliver a routed message

### DISCUSSION THREAD

**Discussion by yslow1987** [rabbitmq/rabbitmq-server#11844] [Questions] [answered]
What are the reason(s) that cause a message published to a classic mirrored queue by fanout exchange but the message is not delivered to connected consumer? The message is gone in the queue and the queue length not yet reached. From the queue statistics from RabbitMQ Management page, the publish count of the queue is increase by 1 but the deliver count is unchanged. There is consumer connected to the queue. Based on the rabbitmq logs, all nodes are working fine and there is no connectivity issue. The message is sent as persistence message and consumer is set as auto acknowledge all message.

**Comment by lukebakken** [comment]
Hello, thanks for using RabbitMQ.

Please read our community support policy - https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md

...especially [this section](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md#who-is-eligible-for-community-support).

> Users who use [the most recent release series](https://www.rabbitmq.com/release-information) and provide _detailed and well researched issue reports_

Your issue sounds like it should be easy to reproduce, so provide a way for us to do so.

Based on your description, your environment has a "bug" - most likely your publisher or consumer is doing something differently than what you think it is.

**Reply by yslow1987** [reply]
Unfortunately it happened once and I have not found a way to reproduce that. Trying my luck trying to ask if anyone face the same issue and can provide some clue to this issue. By the way, the RabbitMQ version that I use is 3.8.9.

**Reply by michaelklishin** [reply]
Which has reached EOL several years ago. Even 3.12 is no longer [covered by community support](https://www.rabbitmq.com/release-information).

**Reply by michaelklishin** [reply]
My guess would be that the CMQ did not have [an elected (or electable) leader](https://www.rabbitmq.com/docs/ha#unsynchronised-mirrors), that should be visible in the logs from all nodes (not at the time of publishing, potentially long before that).

**Reply by michaelklishin** [reply]
Time to [upgrade](https://www.rabbitmq.com/docs/upgrade) from 3.8 and from classic mirrored queues. For a six series jump, the best bet is likely not a series of in-place upgrades but rather a [Blue-Green deployment upgrade](https://www.rabbitmq.com/docs/blue-green-upgrade), potentially with a [default queue type](https://www.rabbitmq.com/docs/vhosts#default-queue-type) configured to simplify the migration.

DQT does not mean that all queues must be quorum queues. That is [absolutely not the case](https://www.rabbitmq.com/docs/quorum-queues#use-cases). Some can remain non-mirrored classic queues but if the majority of queues on 3.8 are mirrored, then setting DQT to `quorum` may be an option.

**Comment by michaelklishin** [comment, marked as answer]
Classic queue mirroring [was removed from RabbitMQ](https://github.com/rabbitmq/rabbitmq-server/pull/9815) for 4.x.

We will no longer investigate anything related to classic mirrored queues. They have been [very visibly](https://www.rabbitmq.com/docs/ha) deprecated for years.

Time to [upgrade](https://www.rabbitmq.com/docs/upgrade) from 3.8 and from classic mirrored queues to quorum queues and/or streams. 

For a six series jump, the best bet is likely not a series of in-place upgrades but rather a [Blue-Green deployment upgrade](https://www.rabbitmq.com/docs/blue-green-upgrade), potentially with a [default queue type](https://www.rabbitmq.com/docs/vhosts#default-queue-type) configured to simplify the migration.

DQT does not mean that all queues must be quorum queues. That is [absolutely not the case](https://www.rabbitmq.com/docs/quorum-queues#use-cases). Some can remain non-mirrored classic queues but if the majority of queues on 3.8 are mirrored, then setting DQT to `quorum` may be an option.
